test/util/testnode/utils.go

```diff
@@ -0,0 +1,119 @@
+package testnode
```

Comment:
I suggest adding descriptions for each function in this file and shedding light on aspects like the outputs are randomized or deterministic (if the function name does not indicate so). This will help other developers understand the purpose and usage of the tools without having to delve deep into the code.

Reply:
This file was moved across from the testfactory package. I didn't want to do this but we have a habit of writing tests using the same package name. As a lot of tests require the encoding config which imports the `app` package and essentially everything else we can often find ourselves caught in a dependency loop. I would prefer if people wrote tests using the `package types_test` instead of `package types` and would want to slowly move the repo to using that convention.

Alternatively, I thought about having the encoding.Config passed as a variable which might still be the better idea